src/System.Windows.Forms.Design/src/System/Windows/Forms/Design/DesignerToolStripControlHost.cs

```diff
@@ -32,7 +31,20 @@ protected override Size DefaultSize
 
     internal GlyphCollection GetGlyphs(ToolStrip parent, GlyphCollection glyphs, Behavior.Behavior standardBehavior)
     {
-        _behaviorService ??= (BehaviorService)parent.Site.GetService(typeof(BehaviorService));
+        if (_behaviorService is null)
+        {
+            if (!parent.Site.TryGetService(out BehaviorService? behaviorService))
+            {
+                throw new InvalidOperationException();
+            }
+
+            _behaviorService = behaviorService;
+        }
```

Comment:
I believe we have `GetRequiredService` available here
```suggestion
        _behaviorService ??= parent.Site.GetRequiredService<BehaviorService>();
```